File: src/11-challenge/solution.ts
export default function decodeFilename(filename: string): string {
	const regex: RegExp = new RegExp(/(?!\d*_)(.*)(?=\..*)/)
	const matchExec: RegExpExecArray | null = regex.exec(filename)
	return matchExec ? matchExec[0] : filename
}

File: src/11-challenge/solution.py
import re


def decode_filename(filename: str) -> str:
    regex: re.Pattern[str] = re.compile(pattern=r"(\d*_)(.*)(\..*)")
    matches: re.Match[str] | None = regex.search(string=filename)
    return matches.group(2) if matches is not None else filename
